nvme.c

```diff
@@ -1184,13 +1189,52 @@ static int create_ns(int argc, char **argv, struct command *cmd, struct plugin *
 		{"flbas", 'f', "NUM", CFG_BYTE,        &cfg.flbas, required_argument, flbas},
 		{"dps",   'd', "NUM", CFG_BYTE,        &cfg.dps,   required_argument, dps},
 		{"nmic",  'm', "NUM", CFG_BYTE,        &cfg.nmic,  required_argument, nmic},
+		{"block-size",   'b', "NUM",  CFG_LONG_SUFFIX, &cfg.bs,        required_argument, bs},
 		{NULL}
 	};
 
 	fd = parse_and_open(argc, argv, desc, command_line_options, &cfg, sizeof(cfg));
 	if (fd < 0)
 		return fd;
 
+	if (cfg.flbas != 0xFF && cfg.bs != 0x00) {
+		fprintf( stderr,
+			"Invalid specification fo both FLBAS and Block Size, please specify only one\n");
+			return EINVAL;
+	}
+
+	if (cfg.bs) {
+		if ((cfg.bs & (~cfg.bs + 1)) != cfg.bs) {
+			fprintf( stderr,
+				"Invalid value for block size (%llu). Block size must be a power of two\n", cfg.bs);
+			return EINVAL;
+		}
+		err = nvme_identify_ns(fd, NVME_NSID_ALL, 0, &ns);
+		if (err) {
+			if (err < 0)
+				perror("identify-namespace");
+			else
+				fprintf(stderr,
+					"NVME Admin command error:%s(%x)\n",
+					nvme_status_to_string(err), err);
+			return err;
+		}
+		for (i=0; i<16; ++i) {
+			if ( (1 << ns.lbaf[i].ds) == cfg.bs && ns.lbaf[i].ms == 0) {
+				cfg.flbas = i;
+				break;
+			}
+		}
+		if (cfg.flbas == 0xff) {
```

Comment:
Just move this check outside this section, and that should fix the issue in the "flbas == 0xff && cfg.bs == 0" issue.